src/measurement_buffer.cpp

```diff
@@ -59,9 +58,9 @@ MeasurementBuffer::MeasurementBuffer(
   const bool & clearing, const double & voxel_size, const Filters & filter,
   const int & voxel_min_points, const bool & enabled,
   const bool & clear_buffer_after_reading, const ModelType & model_type,
-  std::shared_ptr<rclcpp_lifecycle::LifecycleNode> node)
+  const nav2_util::LifecycleNode::WeakPtr & parent)
```

Comment:
Same here, we can update the constructor to give the clock / logger so that this doesn't need the node at all.